packages/types/src/AssetHash.ts

```diff
@@ -7,15 +7,21 @@ export interface AssetHash extends String {
 }
 
 export function AssetHash(value: string) {
-  if (!value.startsWith('0x')) {
-    value = '0x' + value
+  if (value.startsWith('0x')) {
+    value = value.slice(2)
```

Comment:
I'm not sure if this is a better approach. We are slicing the `0x` prefix if it's there and then we are adding it anyway.

Reply:
This is done to pad the length to 64. So that 0x000ab and 0xab end up being equal.

Reply:
Ok, makes sense then